modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenModel.java

```diff
@@ -58,7 +58,7 @@
     public Set<String> allMandatory;
 
     public Set<String> imports = new TreeSet<String>();
-    public boolean hasVars, emptyVars, hasMoreModels, hasEnums, isEnum, hasRequired, hasOptional, isArrayModel, hasChildren;
+    public boolean hasVars, emptyVars, hasMoreModels, hasEnums, isEnum, hasRequired, hasOptional, isArrayModel, hasChildren, isCollectionModel;
```

Comment:
(I may be wrong as I'm new to aspnetcore)

How about add `isMapModel` instead of `isCollectionModel`?

I think `isCollectionModel` is a bit complicated: It is ArraySchema and also MapSchema.


To keep our project as simple as possible, I think `isMapModel` may be better.

For template creator:

- in case of treating ArraySchema => `{{isArrayModel}}`
- in case of treating MapSchema => `{{isMapModel}}`
- in case of treating something collection => `{{isArrayModel}}` & `{{isMapModel}}`

---

Assuming that `isMapModel` exists, it is a good idea to add `isCollectionModel` as it can be possible handling some collection models  transparently. But if we can satify our requirements without it for now, we need to postpone.